Xplat/src/main/java/vazkii/patchouli/common/book/Book.java

```diff
@@ -158,7 +159,7 @@ public Book(JsonObject root, XplatModContainer owner, ResourceLocation id, boole
 		if (noBook) {
 			// Parse on load to catch errors, but need lazy loading for mods
 			// that load after Patchouli
-			var parsed = ItemStackUtil.parseItemStackString(customBookItem);
+			var parsed = ItemStackUtil.deserializeStack(customBookItem, VanillaRegistries.createLookup());
```

Comment:
this does an itemstack lookup and also could have dynamic registries in the components (see enchantments in 1.21)

Reply:
Ok I checked and this runs way before there are dynamic registries (which is a problem for another time)